### PR TITLE
provider/aws: Wait for Spot Fleet to drain before removing from state

### PR DESCRIPTION
**Pull request by catsby**
Ensures the spot fleet is drained before reporting successful destroy
and moving on. We found our nightly acc. tests were leaking Spot Instances (specifically `TestAccAWSSpotFleetRequest_withWeightedCapacity`), and on investigation found that on `DELETE`, `aws_spot_fleet` was not waiting for the included instances to terminate before moving on. 

The problem then appears because Terraform moves on and destroys the IAM role that was being used by the Spot Fleet Request. By the time EC2 gets around to attempting to destroy the request and it's instances, the IAM role has been destroyed, so then we get a failure in the Spot Request. By then, Terraform has likely already exited (or is at least considered it's job done there).

This PR waits for the instances to drain before moving on. Unfortunately the `ActiveInstances` attribute is documented to "possibly be out of date". In my findings it lags about 2 minutes behind.

Alternative I tried simply checking the spot fleet status and looping until we go from `cancelled_terminating` to `cancelled`, however, I found that transition takes upwards of 10 some minutes after all the instances are terminated. 

Unfortunately (still), there exists an edge case in our tests where EC2 is lagging behind in fulfilling the request, such that when we try to destroy it we have zero `ActiveInstances`, so we think things are fine. I'm open to ideas on how to fix that, but in the meantime I believe this PR is still a good idea and will prevent `TestAccAWSSpotFleetRequest_withWeightedCapacity` from leaking ~90% of the time.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_spot_fleet_request.go [hunk 1]
```diff
@@ -945,7 +945,36 @@ func resourceAwsSpotFleetRequestDelete(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error cancelling spot request (%s): %s", d.Id(), err)
 	}
 
-	return nil
+	// check response successfulFleetRequestSet to make sure our request was canceled
+	var found bool
+	for _, s := range resp.SuccessfulFleetRequests {
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_spot_fleet_request.go]: Nice work here checking the `successfulFleetRequestSet` :)

File: builtin/providers/aws/resource_aws_spot_fleet_request.go [hunk 2]
```diff
@@ -945,7 +945,36 @@ func resourceAwsSpotFleetRequestDelete(d *schema.ResourceData, meta interface{})
...
+		if *s.SpotFleetRequestId == d.Id() {
+			found = true
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("[ERR] Spot Fleet request (%s) was not found to be successfully canceled, dangling may resources exit", d.Id())
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_spot_fleet_request.go]: `dangling may resources exit` - this doesn't make any sense

Should it be `dangling resources may resist`?

**Comment by stack72** [comment]
1 small nit pick - apart from that, LGTM! 

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSSpotFleetRequest_'                                                                 130 ↵
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/09/22 21:03:10 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSSpotFleetRequest_ -timeout 120m
=== RUN   TestAccAWSSpotFleetRequest_changePriceForcesNewRequest
--- PASS: TestAccAWSSpotFleetRequest_changePriceForcesNewRequest (121.37s)
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzOrSubnetInRegion (64.22s)
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceAzInGivenList (68.86s)
=== RUN   TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList
--- PASS: TestAccAWSSpotFleetRequest_lowestPriceSubnetInGivenList (87.79s)
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameAz (71.47s)
=== RUN   TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet
--- PASS: TestAccAWSSpotFleetRequest_multipleInstanceTypesInSameSubnet (90.67s)
=== RUN   TestAccAWSSpotFleetRequest_overriddingSpotPrice
--- PASS: TestAccAWSSpotFleetRequest_overriddingSpotPrice (74.57s)
=== RUN   TestAccAWSSpotFleetRequest_diversifiedAllocation
--- PASS: TestAccAWSSpotFleetRequest_diversifiedAllocation (148.58s)
=== RUN   TestAccAWSSpotFleetRequest_withWeightedCapacity
--- PASS: TestAccAWSSpotFleetRequest_withWeightedCapacity (157.40s)
=== RUN   TestAccAWSSpotFleetRequest_CannotUseEmptyKeyName
--- PASS: TestAccAWSSpotFleetRequest_CannotUseEmptyKeyName (0.00s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/aws   	884.938s
```

File: builtin/providers/aws/resource_aws_spot_fleet_request.go (hunk 2 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_spot_fleet_request.go]: 😱

Comment (catsby) [inline builtin/providers/aws/resource_aws_spot_fleet_request.go]: fixed in 1c2a783
